.github/workflows/backwards_compatibility_tests_workflow.yml

```diff
@@ -63,11 +105,43 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Install dependencies
+      - if: startsWith(matrix.os,'ubuntu')
+        name: Install dependencies on ubuntu
         run: |
           sudo apt-get install libopenblas-dev gfortran -y
 
-      - name: Run k-NN Rolling-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }}
+      - if: startsWith(matrix.os,'windows')
+        name: Install MinGW Using Scoop on Windows
+        run: |
+          iex "& {$(irm get.scoop.sh)} -RunAsAdmin"
+          scoop bucket add main
+          scoop install mingw
+
+      - if: startsWith(matrix.os,'windows')
+        name: Add MinGW to PATH on Windows
+        run: |
+          echo "C:/Users/runneradmin/scoop/apps/mingw/current/bin" >> $env:GITHUB_PATH
+          refreshenv
+
+      - if: startsWith(matrix.os,'windows')
+        name: Download OpenBLAS on Windows
+        run: |
+          curl -L -O https://github.com/xianyi/OpenBLAS/releases/download/v0.3.21/OpenBLAS-0.3.21-x64.zip
+          mkdir OpenBLAS
+          Expand-Archive -Path .\OpenBLAS-0.3.21-x64.zip -DestinationPath .\OpenBLAS\
+          mkdir ./src/main/resources/windowsDependencies
```

Comment:
question: Why are we copying it to the "src/main/resources" folder? It seems like a tmp folder in the root or build root would make more sense.

Reply:
We are using src/main/resources/windowsDependencies to copy windows dependencies (dlls) just to keep it uniform with local development environment. If we copy it to temp directory like build, gradle clean will delete it and we might need to copy them again and again while testing.